buildSrc/src/test/java/org/elasticsearch/gradle/NoticeTaskTest.java

```diff
@@ -0,0 +1,165 @@
+package org.elasticsearch.gradle;
+
+import org.elasticsearch.gradle.test.GradleUnitTestCase;
+import org.gradle.api.Project;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.gradle.testfixtures.ProjectBuilder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.rules.TemporaryFolder;
+
+import static java.util.Collections.singletonMap;
+import static org.apache.commons.io.FileUtils.readFileToString;
+import static org.apache.commons.io.FileUtils.write;
+import static org.junit.Assert.*;
+
+public class NoticeTaskTest extends GradleUnitTestCase {
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
```

Comment:
The Gradle test project creates it's own temporary folder no need to pass it in.